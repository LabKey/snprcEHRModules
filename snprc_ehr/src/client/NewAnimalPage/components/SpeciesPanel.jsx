import React from 'react';
import FormGroup from 'react-bootstrap/lib/FormGroup';
import { Col, Row, Radio, Pager, Label } from 'react-bootstrap';
import Select from 'react-select';

export default class SpeciesPanel extends React.Component {


    handleAcquisitionOptionChange = e => {
        const option = e.target.value;
        this.props.handleAcquisitionOptionChange(option);
        this.props.handleLoadAcuisitionTypes(option);
    }
    handleSpeciesChange = value => {
        this.props.handleSpeciesChange(value);
    }
    render() {
        return (
            <div className="wizard-panel__rows">
                <div className='wizard-panel__row' >
                    <Col md={4}>
                    
                        <label  className="field-label" >Acquisition Type</label>
                        <FormGroup id="acquisition-radio-group">
                            <Radio inline className='species-radio' name='acqType' value='Birth'
                                checked={this.props.selectedOption === 'Birth'}
                                onChange={this.handleAcquisitionOptionChange}>
                                Birth
                            </Radio>
                            <Radio inline className='species-radio' name='acqType' value='Acquisition'
                                checked={this.props.selectedOption === 'Acquisition'}
                                onChange={this.handleAcquisitionOptionChange}>
                                Other Acquisition
                            </Radio>
                        </FormGroup>
                    </Col>
                
                    <Col md={4} mdOffset={1}>
                        <label  className="field-label" >Species</label>
                        <Select
                            className="shared-dropdown shared-dropdown__species_width"
                            classNamePrefix="shared-select"
                            options={this.props.speciesList}
                            onChange={this.handleSpeciesChange}
                            placeholder="Select Species"
                            isDisabled={false}
                            id="species-select"
                        />
                    </Col>
                </div>
                
            </div>
        )
    }
}
