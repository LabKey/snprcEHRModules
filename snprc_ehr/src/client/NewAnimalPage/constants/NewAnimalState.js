import moment from 'moment';

const NewAnimalState = () => {

    return  ({
        currentPanel: 1,
        selectedOption: undefined,
        newAnimalData: {
            selectedOption: undefined,
            id: undefined,
            birthDate: { date: moment() },
            bdStatus: undefined,
            acquisitionType: undefined,
            acqDate: { date: moment() },
            gender: undefined,
            sire: undefined,
            dam: undefined,
            species: undefined,
            colony: undefined,
            animalAccount: undefined,
            ownerInstitution: undefined,
            responsibleInstitution: undefined,
            room: undefined,
            cage: { value: undefined },
            diet: undefined,
            pedigree: undefined,
            iacuc: undefined
        },
        summaryData: [],
        speciesList: [],
        acquisitionTypeList: [],
        potentialDamList: [],
        potentialSireList: [],
        locationList: [],
        accountList: [],
        colonyList: [],
        dietList: [],
        institutionList: [],
        iacucList: [],
        pedigreeList: [],
        bdStatusList: [],
        isLoading: true,
        hasError: false,
        preventNext: true,
        saveOk: false,
        showSaveModal: false,
        showCancelModal: false,
        showSpeciesChangeModal: false,
        errorMessage: undefined
    });
}

export default NewAnimalState;
