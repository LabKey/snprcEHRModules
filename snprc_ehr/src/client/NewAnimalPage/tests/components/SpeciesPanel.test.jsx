import React from 'react';
import { shallow } from 'enzyme';
import SpeciesPanel from '../../components/SpeciesPanel';
import NewAnimalState from '../../constants/NewAnimalState';
import lists from '../fixtures/lists';

test("Should render the SpeciesPanel", () => {
  const { newAnimalData, selectedOption} = new NewAnimalState();

  const wrapper = shallow (
    <SpeciesPanel
        handleAcquisitionOptionChange={() => { return false; }}
        handleSpeciesChange={() => { return false; }}
        handleLoadAcuisitionTypes={() => { return false; }}
        speciesList={lists.speciesList}
        selectedOption={selectedOption}
        disabled={false}
        newAnimalData={newAnimalData}
    />
  );
    expect(wrapper).toMatchSnapshot();
});
