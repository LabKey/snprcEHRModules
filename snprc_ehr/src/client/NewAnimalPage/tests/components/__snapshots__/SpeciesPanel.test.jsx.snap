// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the SpeciesPanel 1`] = `
<div
  className="wizard-panel__rows"
>
  <div
    className="wizard-panel__row"
  >
    <div
      className="wizard-panel__col radio-div"
    >
      <label
        className="field-label-align-top"
      >
        Acquisition Type
      </label>
      <FormGroup
        bsClass="form-group"
        disabled={false}
        id="acquisition-radio-group"
      >
        <Radio
          bsClass="radio"
          checked={false}
          className="species-radio"
          disabled={false}
          inline={true}
          name="acqType"
          onChange={[Function]}
          title=""
          value="Birth"
        >
          Birth
        </Radio>
        <Radio
          bsClass="radio"
          checked={false}
          className="species-radio"
          disabled={false}
          inline={true}
          name="acqType"
          onChange={[Function]}
          title=""
          value="Acquisition"
        >
          Other Acquisition
        </Radio>
      </FormGroup>
    </div>
    <div
      className="wizard-panel__col"
    >
      <label
        className="field-label-align-close"
      >
        Species
      </label>
      <StateManager
        className="shared-dropdown shared-dropdown__species_width"
        classNamePrefix="shared-select"
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        id="species-select"
        isDisabled={false}
        isLoading={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "arcSpeciesCode": "PC",
              "id": 10,
              "label": "PCA (PC) - Papio hamadryas anubis/Baboon",
              "value": "PCA",
            },
            Object {
              "arcSpeciesCode": "MM",
              "id": 11,
              "label": "MML (MM) - Macaca mulatta/Rhesus",
              "value": "MML",
            },
          ]
        }
        placeholder="Select Species"
      />
    </div>
  </div>
</div>
`;
